File: docs/APIs-and-SDKs/SDK-Documentation/getting-started/custom-event-logger/flutter/customEventLogger.dart
class CustomEventLogger implements ContextEventLogger {
  @override
  void handleEvent(Context context, EventType event, dynamic data) {
    switch (event) {
      case EventType.Exposure:
        final Exposure exposure = data;
        print("exposed to experiment ${exposure.name}");
        break;
      case EventType.Goal:
        final GoalAchievement goal = data;
        print("goal tracked: ${goal.name}");
        break;
      case EventType.Error:
        print("error: $data");
        break;
      case EventType.Publish:
      case EventType.Ready:
      case EventType.Refresh:
      case EventType.Close:
        break;
    }
  }
}

File: docs/APIs-and-SDKs/SDK-Documentation/getting-started/custom-event-logger/android/customEventLogger.kt
class CustomEventLogger : ContextEventLogger {
    override fun handleEvent(context: Context, event: ContextEventLogger.EventType, data: Any?) {
        when (event) {
            ContextEventLogger.EventType.Exposure -> {
                val exposure = data as Exposure
                println("exposed to experiment ${exposure.name}")
            }
            ContextEventLogger.EventType.Goal -> {
                val goal = data as GoalAchievement
                println("goal tracked: ${goal.name}")
            }
            ContextEventLogger.EventType.Error -> {
                println("error: $data")
            }
            ContextEventLogger.EventType.Publish,
            ContextEventLogger.EventType.Ready,
            ContextEventLogger.EventType.Refresh,
            ContextEventLogger.EventType.Close -> {}
        }
    }
}

val sdkConfig = ABsmartlyConfig.create()
sdkConfig.setContextEventLogger(CustomEventLogger())

val contextConfig = ContextConfig.create()
contextConfig.setEventLogger(CustomEventLogger())
